Add Zhihu data (#1459)

wangrui6 opened the pull request:
Adds Zhihu selected KOL data

Issue: #1459

wangrui6 commented:
Data scraping is still in progress and I plan to regularly update the Huggingface Dataset card here: https://huggingface.co/datasets/wangrui6/Zhihu-KOL

olliestanley commented:
Is there a reason to include both the notebook and the main script? It seems to be the same code in two formats. I would suggest excluding the notebook from the repo for cleanliness

wangrui6 commented:
any update on this PR?

wangrui6 commented:
> Is there a reason to include both the notebook and the main script? It seems to be the same code in two formats. I would suggest excluding the notebook from the repo for cleanliness

deduped notebook and merged with mainline.

Vechtomov commented:
Looks like something is wrong with your main merge. Can you recreate the pull request?

wangrui6 commented:
> Looks like something is wrong with your main merge. Can you recreate the pull request?

Rebased and squashed all the commits. Conflicts are also resolved.

wangrui6 commented:
@olliestanley @Vechtomov Can you review and leave any more comments if any?

bitplane commented on data/datasets/zhihu-kol/main.py:
```diff
@@ -0,0 +1,244 @@
+    limit = 20
+    dfs: list[pd.DataFrame] = []
+    url = f"https://api.zhihu.com/members/{uid}/answers"
+    while 1:
```
ought to be `while True`

wangrui6 replied:
will change it in the next PR.

bitplane commented:
I haven't run it myself yet, but the code looks very nice. Well structured, isolated and documented, very readable. Nice work 🙂👍

Vechtomov reviewed (changes requested):
Can you reupload the dataset? Looks like now `load_dataset("wangrui6/Zhihu-KOL")` loads only first datafile.
![image](https://user-images.githubusercontent.com/28014244/222971422-6c1eb140-5e6a-4565-a99f-1832cb9c6686.png)

wangrui6 commented:
> Can you reupload the dataset? Looks like now `load_dataset("wangrui6/Zhihu-KOL")` loads only first datafile. ![image](https://user-images.githubusercontent.com/28014244/222971422-6c1eb140-5e6a-4565-a99f-1832cb9c6686.png)

Fixed and merged into one file.

Question: In future, do we allow to load multiple parquet files? That will help people to manage the data version and not necessarily need to join data multiple times when there is new data obtained from crawlers. What do you think? 

Suggestion: allow `load_dataset("wangrui6/Zhihu-KOL")` to load multiple files in the data training pipeline.

Vechtomov commented:
> > Can you reupload the dataset? Looks like now `load_dataset("wangrui6/Zhihu-KOL")` loads only first datafile. ![image](https://user-images.githubusercontent.com/28014244/222971422-6c1eb140-5e6a-4565-a99f-1832cb9c6686.png)
> 
> Fixed and merged into one file.
> 
> Question: In future, do we allow to load multiple parquet files? That will help people to manage the data version and not necessarily need to join data multiple times when there is new data obtained from crawlers. What do you think?
> 
> Suggestion: allow `load_dataset("wangrui6/Zhihu-KOL")` to load multiple files in the data training pipeline.

`load_dataset` load all files by default. It seems that format of your first file broke this feature. Check this: https://huggingface.co/docs/datasets/repository_structure

bitplane reviewed (approved):
Looks good to me

wangrui6 commented:
Question, can we still constantly update the HuggingFace data card as we are crawling more and more data every day? Or should we create a new PR to give a different pointer in the data card list?

Vechtomov commented:
> Question, can we still constantly update the HuggingFace data card as we are crawling more and more data every day? Or should we create a new PR to give a different pointer in the data card list?

Yes, you can update the dataset without new PR.

wangrui6 commented:
Update: Over 1 million highly voted question answer pairs have been uploaded to Huggingface https://huggingface.co/datasets/wangrui6/Zhihu-KOL/tree/main/data
